starter-code/src/memory.js

```diff
@@ -1,15 +1,40 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+var MemoryGame = function(cards) {
+  this.cards = cards;
+  this.pickedCards = [];
+  this.pairsClicked = 0;
+  this.pairsGuessed = 0;
+};
 
-// MemoryGame.prototype.shuffleCard = function (cardsArr) {
+MemoryGame.prototype.shuffleCard = function(cardsArr) {
+  var m = cardsArr.length,t,i;
 
-// };
+  // While there remain elements to shuffle…
+  cardsArr.forEach(function() {
+    // Pick a remaining element…
+    i = Math.floor(Math.random() * m--);
 
-// MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
+    // And swap it with the current element.
+    t = cardsArr[m];
+    cardsArr[m] = cardsArr[i];
+    cardsArr[i] = t;
+  });
+  return cardsArr;
+};
 
```

Comment:
MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
  this.pairsClicked++;
  if (firstCard === secondCard) {
    memoryGame.pairsGuessed++;
  }
  return firstCard === secondCard;
}